### PR TITLE
Add circle ci config

### PR DESCRIPTION
**Pull request by vincerubinetti**

### CODE REVIEW

**Comment by dhimmel** [comment]
Noting the circle config we used an an example at https://github.com/greenelab/hetmech-frontend/blob/master/.circleci/config.yml

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            Pushed by $CIRCLE_BUILD_URL
+            [skip ci]"
+workflows:
+  version: 2
+  build:
+    jobs:
+      - deploy:
+          filters:
+            branches:
+              only: master
```
Comment (dhimmel) [inline .circleci/config.yml]: final newline

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+version: 2
+
+jobs:
+  deploy:
```
Comment (dhimmel) [inline .circleci/config.yml]: The problem with only having a deploy job on master is that pull requests are not going to get built. I think it's nice to have PR builds.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          fingerprints:
+            - "63:a6:74:80:69:f1:6d:86:c0:97:24:bb:99:1c:7f:fb"
+      - run:
+          name: Deploy build to build branch
+          command: |
+            echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+            build --dist build \
+              --user "CircleCI Deploy Job <ci-build@circleci.com>" \
+              --message \
+            "Deploy webpage on `date --utc --iso-8601`
```
Comment (dhimmel) [inline .circleci/config.yml]: Should "webpage" be changed to "node package" or whatever the build is?

File: .circleci/config.yml (hunk 2 above)
Comment (vincerubinetti) [reply .circleci/config.yml]: Got rid of the filter. Is the new syntax I have right?

Comment (dhimmel) [reply .circleci/config.yml]: The problem now is that PR builds are going to attempt to deploy. Deployment must be conditional upon it being in the master branch of `hetio/frontend-components`

**Comment by dhimmel** [comment]
https://circleci.com/setup-project/gh/hetio/frontend-components is taking a long time to load. Once it loads, I think I will be able to click "Start building". Then hopefully it will build any new commits on this PR.

**Comment by dhimmel** [comment]
Okay, https://circleci.com/gh/hetio/frontend-components is now up. @vincerubinetti  can you make a new commit or do `git commit --amend --no-edit` and force push so we can trigger CI

File: .circleci/config.yml [hunk 4]
```diff
@@ -19,15 +19,12 @@ jobs:
             build --dist build \
               --user "CircleCI Deploy Job <ci-build@circleci.com>" \
               --message \
-            "Deploy webpage on `date --utc --iso-8601`
+            "Deploy npm package on `date --utc --iso-8601`
```
Comment (dhimmel) [inline .circleci/config.yml]: This change was reverted in https://github.com/hetio/frontend-components/pull/6/commits/fc17f1e87dad30e518dcb85902e311cc42c41d82

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          command: |
+            npm install
+      - add_ssh_keys:
+          fingerprints:
+            - "63:a6:74:80:69:f1:6d:86:c0:97:24:bb:99:1c:7f:fb"
+      - run:
+          name: Deploy build to build branch
+          command: |
+            echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+            build --dist build \
```
Comment (dhimmel) [inline .circleci/config.yml]: This line is wrong. See https://github.com/greenelab/hetmech-frontend/blob/25224764c3509129c9a3132980d9f56b6f81ae0b/.circleci/config.yml#L42.

Docs for https://www.npmjs.com/package/gh-pages

```
Usage: gh-pages [options]

Options:
  -V, --version            output the version number
  -d, --dist <dist>        Base directory for all source files
  -s, --src <src>          Pattern used to select which files to publish (default: "**/*")
  -b, --branch <branch>    Name of the branch you are pushing to (default: "gh-pages")
  -e, --dest <dest>        Target directory within the destination branch (relative to the root) (default: ".")
  -a, --add                Only add, and never remove existing files
  -x, --silent             Do not output the repository url
  -m, --message <message>  commit message (default: "Updates")
  -g, --tag <tag>          add tag to commit
  -t, --dotfiles           Include dotfiles
  -r, --repo <repo>        URL of the repository you are pushing to
  -p, --depth <depth>      depth for clone (default: 1)
  -o, --remote <name>      The name of the remote (default: "origin")
  -u, --user <address>     The name and email of the user (defaults to the git config).  Format is "Your Name <email@example.com>".
  -v, --remove <pattern>   Remove files that match the given pattern (ignored if used together with --add). (default: ".")
  -n, --no-push            Commit only (with no push)
  -h, --help               output usage information
```

We will need to set `--branch build`

**Comment by dhimmel** [comment]
I am going to edit the master branch to squash the commits.

**Pull request merged**